### ISSUE TITLE
TZ=Pacific/Auckland python2.4 ./bin/trial twisted.conch.test.test_cftp.ListingTests fails

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3551]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)| @mwhudson reported|
|-|-|
|Trac ID|trac#3551|
|Type|defect|
|Created|2008-11-20 08:44:57Z|
|Branch|https://github.com/twisted/twisted/tree/timezones-are-impossible-3551|

There seems to be a timezone dependency in the new tests for lsLine.  They fail for me here in UTC+13, for example.

<details><summary>Searchable metadata</summary>

```
trac-id__3551 3551
type__defect defect
reporter__mwh mwh
priority__normal normal
milestone__ 
branch__branches_timezones_are_impossible_3551 branches/timezones-are-impossible-3551
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1227170697000000 1227170697000000
changetime__1227551405000000 1227551405000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [25464]) Branching to 'timezones-are-impossible-3551'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Fixed, build results http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/timezones-are-impossible-3551

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_3"></a>|@mwhudson commented|
|-|-|

Looks fine, and indeed passes here (in the future!).  Please land.

Thanks for fixing this.

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)|@mwhudson set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [25489]) Merge timezones-are-impossible-3551

Author: exarkun
Reviewer: mwhudson
Fixes: [#3551](https://github.com/twisted/twisted/issues/3551)

Change the unit tests for `twisted.conch.ls.lsLine` so to set the current timezone for
the process so that the results are more easily predictable.  This fixes the unit tests
in timezones where the values used in the test fall on a different local day than the
tests expected.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
